untie/src/untie.cpp

```diff
@@ -18,36 +19,61 @@ Untie::Untie() {
     log::Logger::setup();
 
     m_engine = std::make_unique<knot::Engine>();
+    Engine::set_active_engine(*m_engine);
+    std::shared_ptr<PxPhysics_ptr_wrapper> px_physics = m_engine->get_physics_module().lock()->get_physics().lock();
```

Comment:
Could this happen in the engine constructor?

Reply:
it saves shader ptr for components such as `RigidBody`.

Reply:
Actually have you tried removing it completely should still work?

Reply:
oh, yeah, I already move it to the on_awake. forget it